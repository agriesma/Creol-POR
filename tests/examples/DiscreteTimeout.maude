load creol-interpreter
mod PROGRAM is
protecting CREOL-SIMULATOR .
op init : -> Configuration [ctor] .
eq init =
< "DiscreteTimeout" : Cl | Inh: noInh, Par: "other", Att: "other" |-> null , "clock" |-> null, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: ( "clock" @@ "DiscreteTimeout" ) ::= int(0) ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: "label:0" |-> null, Code: "clock" ::= "+" ( "clock" :: int(1) ) ; ( "label:0" ! "this" . "run" (emp ) )  ; free( "label:0" ) ; return ( emp ) > *
  < "timeout" : Mtdname | Param: "n", Latt: "n" |-> null , "result" |-> null , "timeout" |-> null , "l" |-> null, Code: "timeout" ::= "+" ( "clock" :: int(10) ) ; "l" ! "other" . "m" ( emp ) ; ( ( await "&&" ( ( "l" ?? ) :: "<=" ( "clock" :: "timeout" ) ) ) ; ( "l" ? ( noVid ) )  ; "result" ::= bool(true) [] ( await ">" ( "clock" :: "timeout" ) ) ; free( "l" ) ; "result" ::= bool(false) ) ; return ( "result" ) >, Ocnt: 0 >

.
endm
