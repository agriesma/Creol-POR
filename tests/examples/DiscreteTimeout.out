==========================================
reduce in PROGRAM : init .
result Class: 
< "DiscreteTimeout" : Cl | Inh: noInh, Par: "other", Att: "clock" |-> null,
    "other" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: ("clock" @@
    "DiscreteTimeout") ::= int(0) ; return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: "label:0" |-> null, Code: "clock" ::=
    "+"("clock" # int(1)) ; "label:0" ! "this" . "run"(emp) ; free("label:0") ;
    return(emp) > *
  < "timeout" : Mtdname | Param: "n", Latt: "l" |-> null, "n" |-> null,
    "result" |-> null, "timeout" |-> null, Code: "timeout" ::= "+"("clock" #
    int(10)) ; "l" ! "other" . "m"(emp) ; (await ">"("clock" # "timeout") ;
    "result" ::= bool(false) ||| await "&&"("l" ?? # "<="("clock" # "timeout"))
    ; "l" ?(noVid) ; "result" ::= bool(true)) ; return("result") >,
  Ocnt: 0 >
Bye.
